### PR TITLE
Fixed `needs_broadcasting` expression in tf.layers.normalization

### PR DESCRIPTION
**Pull request by sugyan**
in Python3, `range()` function doesn't create lists, so `[0, 1, 2] != range(4)[:-1]` always returns `True`.

### CODE REVIEW

**Comment by martinwicke** [comment]
Thanks for this fix. Can you add a test that would have failed? Clearly this code isn't tested well.

**Review by martinwicke — CHANGES_REQUESTED** [review]
Please add a test for this behavior.

**Comment by sugyan** [comment]
I am sorry for a miscellaneous pull-request with no test code.
I just pointed out a programming mistake.

I read the code and checked its behavior, but in `normalization.py`,` axis` in initializer is a single integer and `len(reduction_axes)` is always equal to `len(input_shape) - 1`, so the calculation results will be exactly the same regardless of the value of `needs_broadcast`...
I'm wondering if this variable is really necessary.

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Review by martinwicke — APPROVED** [review]

**Comment by martinwicke** [comment]
Fine. This is a clear improvement. Thanks!
